connectors/katalog/pkg/connector/connector.go

```diff
@@ -57,3 +72,90 @@ func (r *Handler) getAssetInfo(c *gin.Context) {
 
 	c.JSON(http.StatusOK, &response)
 }
+
+func (r *Handler) reportError(c *gin.Context, httpCode int, errorMessage string) {
+	r.log.Error().Msg(errorMessage)
+	c.JSON(httpCode, gin.H{"error": errorMessage})
+}
+
+// Enables writing of assets to katalog. The different flows supported are:
+// (a) When DestinationAssetID is specified:
+//     Then a destination asset id is created with name : <DestinationAssetID>
+// (b) When DestinationAssetID is not specified but ResourceMetadata.Name of source asset is specified:
+//     Then an asset is created with name: <ResourceMetadata.Name>-<Kubernetes Generated Random String>
+// (c) When DestinationAssetID and ResourceMetadata.Name of source asset are not specified:
+//     Then an asset is created with name: fybrik-<Kubernetes Generated Random String>
+func (r *Handler) createAssetInfo(c *gin.Context) {
+	// Parse request
+	var request datacatalog.CreateAssetRequest
+	if err := c.ShouldBindJSON(&request); err != nil {
+		r.reportError(c, http.StatusBadRequest, "Error during ShouldBindJSON in createAssetInfo"+err.Error())
+		return
+	}
+
+	logging.LogStructure("CreateAssetRequest object received:", request, &r.log, false, false)
+
+	if request.DestinationCatalogID == "" {
+		r.reportError(c, http.StatusBadRequest, "Invalid DestinationCatalogID in request")
+		return
+	}
+
+	var asset *v1alpha1.Asset
```

Comment:
you don't have explicitly define `asset` and `err`
just write `asset := &v1alpa1.Asset {...}`